src/main/java/io/swagger/oas/inflector/controllers/OpenAPIOperationController.java

```diff
@@ -607,14 +665,28 @@ public void setContentType(RequestContext res, ResponseContext resp, Operation o
 
                 for(String key : content.keySet()) {
                     MediaType mediaType = MediaType.valueOf(key);
-                    for (MediaType acceptable : res.getAcceptableMediaTypes()) {
-                        if (mediaType.isCompatible(acceptable)) {
-                            resp.setContentType(mediaType);
-                            return;
+                    if (res.getHeaders().get("Content-Type")!= null) {
+                        for (String acceptable : res.getHeaders().get("Content-Type")) {
+                            String subtype = acceptable.substring(acceptable.lastIndexOf("/") + 1);
+
+                            if (subtype.equals(mediaType.getSubtype())) {
+                                resp.setContentType(mediaType);
+                                return;
+                            }else {
+                                resp.setContentType(mediaType);
+                            }
                         }
-                        else{
-                            resp.setContentType(mediaType);
+                    }else {
+                        for (MediaType acceptable : res.getAcceptableMediaTypes()){
+
+                            if (mediaType.isCompatible(acceptable)) {
+                                resp.setContentType(mediaType);
```

Comment:
same here.

Reply:
It doesn't make sense!. let me check what was I thinking back then.

Reply:
Done. removed the redundant code